bindings/python/fsm.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "common.h"
+
+#include <memory>
+
+class Fsm {
+public:
+    Fsm() { }
+
+#ifndef SWIG
```

Comment:
This is to hide some parts of C++ code from SWIG. These parts are needed only for wrapper classes to communicate with each other and should not be available from target language (Python). In some cases SWIG would emit an error or warning message while parsing this code.

In most cases same result could indeed be achieved with SWIG directives such as `%ignore`, but apparently these directives do not always work well with template classes when used outside of class declaration.

Moreover, IMHO the fact that some declarations are to be ignored should be very explicit. Using SWIG directive would force us to either use `#ifdef SWIG`, or to place the masking directives in the separate `.i` file, which IMHO would be too far from the declaration.